Lib/ast.py

```diff
@@ -62,11 +62,13 @@ def literal_eval(node_or_string):
         node_or_string = parse(node_or_string, mode='eval')
     if isinstance(node_or_string, Expression):
         node_or_string = node_or_string.body
+    def _malformed(node):
+        raise ValueError(f'malformed node or string: {node!r}')
     def _convert_num(node):
         if isinstance(node, Constant):
             if type(node.value) in (int, float, complex):
                 return node.value
-        raise ValueError('malformed node or string: ' + repr(node))
+        _malformed(node)
```

Comment:
Suggestion: I suppose at this point we can invert the check to make the intent a bit more clear:

```
        if not isinstance(node, Constant) or type(node.value) not in (int, float, complex):
            _raise_malformed_node(node)
        return node.value
```